Improvements to Active Sales and New Sale

caike opened the pull request:
Improvements to Active Sales display and New Sale form. 

Active Sales page:

1. Added a fix to the link to ce explorer. Now linking to the policy page instead of directly to the asset. The reason for this is that I'm having a hard time generating the fingerprint value for the asset. It appears [the JS library](https://github.com/Emurgo/cip14-js) which does that is not compatible with Deno/ES modules. In this example, the links go [here](https://preview.cexplorer.io/policy/248c72ada6de918094d07593148bb357f22be94aa52a84f4107dfb7c) and [here](https://preview.cexplorer.io/policy/24b92a0fa3bca70bd659a424c60b86d565bbdc6ebce50fa96bed588a)

2. Displaying asset name in text instead of hex value. In this example, "BananaNFT" and "JungleNFTT1" instead of their hex values.

<img width="478" alt="PicoSwap" src="https://user-images.githubusercontent.com/10722/202298032-60d7f85b-9908-4d36-b412-1533663d04a4.png">


For Sale form:

1. Displaying hex value for the policy ID instead of its bech32 version. It appears the former is more frequently used so this made more sense, but please let me know if that is not the case.

3. Displaying text version of token name instead of hex for better readability.

<img width="608" alt="PicoSwap" src="https://user-images.githubusercontent.com/10722/202299580-5bef297b-4dca-4b1d-9188-424503655069.png">

caike committed:
Fix Overview link to policy and display asset names

caike committed:
Formatting

caike commented on utils.js:
```diff
@@ -0,0 +1,30 @@
+    }else{
+        return "";
+    }
+}
```
Unsure if the function above is the best way to get to the text version of an asset name 🤔

christianschmitz replied:
You could have a look at the `bytesToString` function in the helios library itself (it might be smart to export it at some point so we wouldn't need to reimplement it in PicoSwap). So it would be something like:
```
function hexToString(hex) {
    try {   
        return (new TextDecoder("utf-8", {fatal: true})).decode((new Uint8Array(helios.hexToBytes(hex))).buffer);
    } catch (e) {
        return hex // fallback in tokenName isn't valid UTF-8 (which is allowed according to Plutus Ledger API)
    }
}
```

I would also fall back to original hex if it isn't valid UTF-8

caike replied:
Thanks. Submitted a [PR to helios](https://github.com/Hyperion-BT/Helios/pull/27) with the suggested change. Once that's in place, I believe the final version of the code will look something like this:


```typescript
import { bytesToString, hexToBytes } from "./helios.js";

function hexToString(hex){
    try {
        return bytesToString(hexToBytes(hex));
    } catch (e) {
        return hex;
    }
}
```

christianschmitz commented on utils.js:
```diff
@@ -0,0 +1,30 @@
+    return `${policyID.substring(0,10)}...${policyID.substring(46)}`;
+}
+
+export { fromHexToText, formatMintingPolicyID };
```
Nice! Exactly what I had in mind since the beginning, but didn't implemented due to laziness

christianschmitz commented on inputs.js:
```diff
@@ -25,16 +26,16 @@ function renderInput(props) {
  * @param {string} id 
  * @param {string} value 
- * @param {string[]} options 
+ * @param {{value: String, label: String}[]} options
```
Coding style: `string` instead of `String`

christianschmitz reviewed (changes requested):
Nice work!

Only the fromHexToText function needs to be changed before this can be merged (see inline comments)

caike committed:
Fix coding style

caike committed:
Simplify asset name resolution.

caike commented on helios.js:
```diff
@@ -1,2 +1,25 @@
+    bytesToText,
+    hexToBytes,
+    highlight,
+} from "https://www.hyperion-bt.org/Helios/helios.js";
```
I think sorting it this way makes this file easier to read and its diff less impacted by future changes.

christianschmitz commented on Overview.js:
```diff
@@ -41,7 +41,7 @@ export function Overview(props) {
             for (const tokenName of tokenNames) {
                 const formattedPolicyID = formatMintingPolicyID(mph);
-                const tokenNameString = fromHexToText(bytesToHex(tokenName));
+                const tokenNameString = bytesToText(tokenName);
```
Remember that `bytesToText` throws an error in case the bytes aren't valid Utf-8. So we should still define a function in `utils.js` that wraps `bytesToText` and falls back to the hex representation of the tokenName in case of invalid Utf-8

caike replied:
Ahh.. that's right! Thanks for the heads up. Just added the `parseTokenName` wrapper function.

christianschmitz commented on inputs.js:
```diff
@@ -220,7 +219,7 @@ export class AdaInput {
                 ${renderSelect(tnId, bytesToHex(this.#tokenName),
                         this.#tokenNames.map(t => {
-                            return {value: bytesToHex(t), label: fromHexToText(bytesToHex(t)) };
+                            return {value: bytesToHex(t), label: bytesToText(t)};
```
Same comment as above

caike committed:
Wrap function to treat error.

bytesToText may throw an error in case the bytes aren't valid Utf-8
fallback to hex value is allowed according to Plutus Ledger API

christianschmitz reviewed (approved):
Nice work. Thanks!
